server/services/alertsService.ts

```diff
@@ -153,6 +134,40 @@ export default class AlertsService {
     }
   }
 
+  private async getPagedAlerts(
+    prisonApiClient: PrisonApiClient,
+    prisonerData: Prisoner,
+    queryParams: AlertsListQueryParams,
+    canUpdateAlert: boolean,
+  ): Promise<PagedList<AlertView> | null> {
+    const pagedAlerts = await prisonApiClient.getAlerts(prisonerData.bookingId, this.mapToApiParams(queryParams))
+
+    const alertContentsForView = pagedAlerts.content.map<AlertView>((alert: Alert) => {
+      const alertUpdatable = canUpdateAlert && alert.active && alert.alertCode !== 'DOCGM'
+      return {
+        ...alert,
+        addMoreDetailsLinkUrl: alertUpdatable
+          ? `/prisoner/${prisonerData.prisonerNumber}/alerts/${alert.alertId}/add-more-details`
+          : null,
+        closeAlertLinkUrl:
+          alertUpdatable && !alert.dateExpires
+            ? `/prisoner/${prisonerData.prisonerNumber}/alerts/${alert.alertId}/close`
+            : null,
+        changeEndDateLinkUrl:
+          alertUpdatable && alert.dateExpires
+            ? `/prisoner/${prisonerData.prisonerNumber}/alerts/${alert.alertId}/change-end-date`
+            : null,
+        addedByFullName: formatName(alert.addedByFirstName, undefined, alert.addedByLastName),
+        expiredByFullName: formatName(alert.expiredByFirstName, undefined, alert.expiredByLastName),
+      }
+    })
+
+    return {
+      ...pagedAlerts,
+      content: alertContentsForView,
+    }
+  }
+
```

Comment:
Refactored this a little to enable return of new object rather than mutated data from the api. This was causing a problem with the tests where we were essentially comparing the output to the output, as the mocked input was being changed into the output, so it was erroneously passing